plugins/modules/vpc_subnet.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+"""This module contains all of the functionality for Linode VPCs."""
+
+from __future__ import absolute_import, division, print_function
+
+from typing import Any, Optional
+
+import ansible_collections.linode.cloud.plugins.module_utils.doc_fragments.vpc_subnet as docs
+from ansible_collections.linode.cloud.plugins.module_utils.linode_common import (
+    LinodeModuleBase,
+)
+from ansible_collections.linode.cloud.plugins.module_utils.linode_docs import (
+    global_authors,
+    global_requirements,
+)
+from ansible_collections.linode.cloud.plugins.module_utils.linode_helper import (
+    filter_null_values,
+    get_resource_safe_condition,
+    handle_updates,
+)
+from ansible_specdoc.objects import (
+    FieldType,
+    SpecDocMeta,
+    SpecField,
+    SpecReturnValue,
+)
+from linode_api4 import VPC, VPCSubnet
+
+SPEC = {
+    "vpc_id": SpecField(
+        type=FieldType.integer,
+        required=True,
+        description=["The ID of the parent VPC for this subnet."],
+    ),
+    "label": SpecField(
+        type=FieldType.string,
+        required=True,
+        description=["This VPC's unique label."],
+    ),
+    "state": SpecField(
+        type=FieldType.string,
+        choices=["present", "absent"],
+        required=True,
+        description=["The state of this token."],
+    ),
+    "ipv4": SpecField(
+        type=FieldType.string,
+        description=["The IPV4 range for this subnet in CIDR format."],
+    ),
+}
+
+
+SPECDOC_META = SpecDocMeta(
+    description=[
+        "Create, read, and update a Linode VPC Subnet.",
+    ],
+    requirements=global_requirements,
+    author=global_authors,
+    options=SPEC,
+    examples=docs.specdoc_examples,
+    return_values={
+        "subnet": SpecReturnValue(
+            description="The VPC in JSON serialized form.",
+            docs_url="TODO",
+            type=FieldType.dict,
+            sample=docs.result_subnet_samples,
+        )
+    },
+)
+
+CREATE_FIELDS = {"label", "ipv4"}
+
+
+class Module(LinodeModuleBase):
+    """Module for creating and destroying Linode VPCS"""
+
+    def __init__(self) -> None:
+        self.module_arg_spec = SPECDOC_META.ansible_spec
+        self.results = {"changed": False, "actions": [], "vpc": None}
+
+        super().__init__(
+            module_arg_spec=self.module_arg_spec,
+        )
+
+    def _create(self, vpc: VPC) -> Optional[VPCSubnet]:
+        params = filter_null_values(
+            {k: v for k, v in self.module.params.items() if k in CREATE_FIELDS}
+        )
+
+        try:
+            return vpc.subnet_create(**params)
+        except Exception as exception:
+            return self.fail(msg="failed to create VPC: {0}".format(exception))
+
+    def _update(self, subnet: VPCSubnet) -> None:
```

Comment:
Should we be able to update VPC subnet `label`? https://docs.google.com/document/d/1BE8LkLRzP2dY4S8eUuW1hWHLyJmUWh7PZSOqE94amno/edit#heading=h.nhueb1yp6ywu

Reply:
Gotcha. Thanks for the explanation!